src/Lean/Elab/Command.lean

```diff
@@ -207,11 +210,25 @@ private def addTraceAsMessages : CommandElabM Unit := do
     traceState.traces := {}
   }
 
-private def elabCommandUsing (s : State) (stx : Syntax) : List CommandElab → CommandElabM Unit
+private def mkInfoTree (elaborator : Name) (stx : Syntax) (trees : Std.PersistentArray InfoTree) : CommandElabM InfoTree := do
+  let ctx ← read
+  let s ← get
+  let scope := s.scopes.head!
+  let tree := InfoTree.node (Info.ofCommandInfo { elaborator, stx }) trees
+  let tree := InfoTree.context {
+    env := s.env, fileMap := ctx.fileMap, mctx := {}, currNamespace := scope.currNamespace, openDecls := scope.openDecls, options := scope.opts
+  } tree
+  if checkTraceOption (← getOptions) `Elab.info then
```

Comment:
Did you try to use `trace[Elab.info] ...`? We now have `MonadTrace` for `CommandElabM`.
It `trace` doesn't work here, let's add a comment explaining why.

Reply:
I only moved this code, but `trace` seems to work fine. Changed.